### PR TITLE
Migrate from pytorch-pretrained-bert to transformers

### PR DESCRIPTION
**Pull request by achyudh**
Migrate BERT and HBERT to pytorch-transformers, along with other dependencies (such as tasks.relevance_transfer) that used pytorch-pretrained-bert. Tested all changes locally.

Addressed issue #45.

### CODE REVIEW

**Review by Ashutosh-Adhikari — DISMISSED** [review]
In general LGTM. Glad we don't have tokenizers anymore. However, can you make sure that the results are consistent as requested on the slack channel @achyudh ?

File: models/hbert/sentence_encoder.py [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
+    def __init__(self, config):
+        super().__init__(config, num_labels=config.num_labels)
         self.bert = BertModel(config)
         self.dropout = nn.Dropout(config.hidden_dropout_prob)
-        self.apply(self.init_bert_weights)
+        self.init_weights()
 
     def forward(self, input_ids, token_type_ids=None, attention_mask=None):
-        _, pooled_output = self.bert(input_ids, token_type_ids, attention_mask, output_all_encoded_layers=False)
+        pooled_output = self.bert(input_ids, token_type_ids, attention_mask)[1]
```
Comment (daemon) [inline models/hbert/sentence_encoder.py]: They changed the argument positions going from `pytorch-transformers` to `transformers`.

File: common/trainers/relevance_transfer_trainer.py [hunk 2]
```diff
@@ -6,47 +6,44 @@
...
     def train_epoch(self):
         for step, batch in enumerate(tqdm(self.train_loader, desc="Training")):
             self.model.train()
 
-            if self.config['model'] in {'BERT-Base', 'BERT-Large', 'HBERT-Base', 'HBERT-Large'}:
+            if self.config['model'] in BERT_MODELS:
                 batch = tuple(t.to(self.config['device']) for t in batch)
                 input_ids, input_mask, segment_ids, label_ids = batch
-                logits = torch.sigmoid(self.model(input_ids, segment_ids, input_mask)).squeeze(dim=1)
+                logits = torch.sigmoid(self.model(input_ids, segment_ids, input_mask)[0]).squeeze(dim=1)
```
Comment (daemon) [inline common/trainers/relevance_transfer_trainer.py]: Here too.

File: common/trainers/bert_trainer.py [hunk 3]
```diff
@@ -45,7 +42,7 @@ def train_epoch(self, train_dataloader):
             self.model.train()
             batch = tuple(t.to(self.args.device) for t in batch)
             input_ids, input_mask, segment_ids, label_ids = batch
-            logits = self.model(input_ids, segment_ids, input_mask)
+            logits = self.model(input_ids, segment_ids, input_mask)[0]
```
Comment (daemon) [inline common/trainers/bert_trainer.py]: Here too.

File: common/evaluators/relevance_transfer_evaluator.py [hunk 4]
```diff
@@ -78,7 +67,7 @@ def get_scores(self, silent=False):
                 label_ids = label_ids.to(self.config['device'])
 
                 with torch.no_grad():
-                    logits = torch.sigmoid(self.model(input_ids, segment_ids, input_mask)).squeeze(dim=1)
+                    logits = torch.sigmoid(self.model(input_ids, segment_ids, input_mask)[0]).squeeze(dim=1)
```
Comment (daemon) [inline common/evaluators/relevance_transfer_evaluator.py]: Here too.

File: common/evaluators/bert_evaluator.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ def get_scores(self, silent=False):
             label_ids = label_ids.to(self.args.device)
 
             with torch.no_grad():
-                logits = self.model(input_ids, segment_ids, input_mask)
+                logits = self.model(input_ids, segment_ids, input_mask)[0]
```
Comment (daemon) [inline common/evaluators/bert_evaluator.py]: Here too.

**Review by daemon — CHANGES_REQUESTED** [review]

File: models/hbert/sentence_encoder.py (hunk 1 above)
Comment (achyudh) [reply models/hbert/sentence_encoder.py]: Thanks for pointing that out. I fixed this issue in 16dd9d9, along with fixing the weight decay when initializing the AdamW optimizer.

**Comment by achyudh** [comment]
> In general LGTM. Glad we don't have tokenizers anymore. However, can you make sure that the results are consistent as requested on the slack channel @achyudh ?

Sorry for leaving this PR open over the past month; I verified the results of both BERT Base and BERT Large on Reuters and I was able to reproduce our previous results. 

One thing to note is that a previous commit e52eff453d4c8330b1a6ae5778ed7726cdac5c1e from May resulted in minor changes compared to the results from April's preprint. However, this is minor and doesn't affect the results of the DocBERT paper.

**Review by daemon — APPROVED** [review]
